### PR TITLE
fix access log config

### LINKED ISSUE #19516: Doesn't seem possible to set accessLogEncoding using installer
**Bug description**
When using `istioctl manifest`, `values.global.proxy.accessLogEncoding` doesn't seem to have an effect

**Expected behavior**
Setting `accessLogEncoding: JSON` changes the format for proxy access logs

**Steps to reproduce the bug**
- Create IstioControlPlane config `control-plane.yaml` like
```yaml
apiVersion: install.istio.io/v1alpha2
kind: IstioControlPlane
spec:
  values:
    global:
      proxy:
        accessLogEncoding: JSON
        accessLogFile: /dev/stdout
```
- `istioctl manifest apply -f control-plane.yaml`
- View istio-proxy logs for e.g. Pilot

**Version**
1.4.2

**How was Istio installed?**
istioctl

### PR DESCRIPTION
**Pull request by sakajunquality**
Fixes istio/istio#19516.

accessLogFormat and accessLogEncoding was missing from the ConfigMap Template

### CODE REVIEW

File: istio-control/istio-discovery/templates/configmap.yaml [hunk 1]
```diff
@@ -29,6 +29,12 @@ data:
     # Set accessLogFile to empty string to disable access log.
     accessLogFile: "{{ .Values.global.proxy.accessLogFile }}"
 
+    # If accessLogEncoding is TEXT, value will be used directly as the log format
+    accessLogFormat: {{ .Values.global.proxy.accessLogFormat | quote }}
+
+    # Set accessLogEncoding to JSON or TEXT to configure sidecar access log
+    accessLogEncoding: {{ .Values.global.proxy.accessLogEncoding }}
```
Comment (sakajunquality) [inline istio-control/istio-discovery/templates/configmap.yaml]: originally from here
https://github.com/istio/istio/blob/master/install/kubernetes/helm/istio/templates/configmap.yaml#L53

**Review by morvencao — COMMENTED** [review]
/ok-to-test

**Comment by howardjohn** [comment]
Just merged a Pr doing the same thing, thanks!

**Comment by sakajunquality** [comment]
@howardjohn thank you for the review anyway!

**Pull request closed without merging**